### PR TITLE
docs glossary & git-comparison: fixup & update bookmark->branch transition

### PR DESCRIPTION
**Pull request by ilyagr**
This fixes up git-comparison.md. The changes to the glossary could be moved to a separate commit/PR.

### CODE REVIEW

File: docs/git-comparison.md [hunk 1]
```diff
@@ -110,13 +111,15 @@ parent.
       <td><code>git clone &lt;source&gt; &lt;destination&gt; [--origin &lt;remote name&gt;]</code></td>
     </tr>
     <tr>
-      <td>Update the local repo with all bookmarks from a remote</td>
+      <!-- TODO: Mention that you might need to do a `jj branch track branch@remote` -->
```
Comment (emilazy) [inline docs/git-comparison.md:114]: I’m not sure where this TODO is from, but it seems outdated.

File: docs/glossary.md [hunk 2]
```diff
@@ -22,13 +22,24 @@ such as the "operation store backend" for storing
...
-A bookmark is a named pointer to a [commit](#commit). They automatically follow
-the commit if it gets [rewritten](#rewrite). Bookmarks are sometimes called
-"named branches" to distinguish them from [anonymous branches](#anonymous-branch).
-They are similar to Git's branches and even more similar to
+A bookmark is a named pointer to a [commit](#commit). They are similar to Git's [branches](#branch) and even more similar to
 [Mercurial's bookmarks](https://wiki.mercurial-scm.org/Bookmarks). See
 [here](bookmarks.md) for details.
 
+Unlike in Git, bookmarks automatically follow the commit if it gets
+[rewritten](#rewrite). However, bookmarks do *not* move when you create a new commit;
```
Comment (emilazy) [inline docs/glossary.md:30]: What are some practical cases where you rewrite a commit in Git and the corresponding branch doesn’t get rewritten? I worry this is a bit misleading as‐is.

**Review by emilazy — COMMENTED** [review]
Broadly LGTM. We might want to explicitly discuss the bookmark–branch distinction in the comparison, but this is an improvement over the current wrong state.

File: docs/git-comparison.md [hunk 3]
```diff
@@ -44,10 +46,9 @@ various use cases.
   updated, and so will the working copy if it points to any of the rebased
   commits.
 * **Branches are identified by their names (across remotes).** For example, if
```
Comment (martinvonz) [inline docs/git-comparison.md:48]: maybe "Branches/bookmarks" here

File: docs/git-comparison.md [hunk 4]
```diff
@@ -324,29 +327,29 @@ parent.
...
     <tr>
-      <td>Move a bookmark backward or sideways</td>
+      <td>Move a bookmark/branch backward or sideways</td>
       <td><code>jj bookmark set &lt;name&gt; -r &lt;revision&gt; --allow-backwards</code></td>
-      <td><code>git bookmark -f &lt;name&gt; &lt;revision&gt;</code></td>
+      <td><code>git branch -f &lt;name&gt; &lt;revision&gt;</code></td>
     </tr>
     <tr>
-      <td>Delete a bookmark</td>
+      <td>Delete a bookmar/branchk</td>
```
Comment (martinvonz) [inline docs/git-comparison.md:350]: fix the typos here

File: docs/git-comparison.md (hunk 1 above)
Comment (ilyagr) [reply docs/git-comparison.md:114]: I just added it :) It's obviously not clear enough, but honestly it's unclear to me what to do here.

The problem is that if you `jj git fetch` from a remote, you'll get all the bookmarks, but you won't see them in `jj log` until you `jj bookmark track` something. I think new people are likely to be confused by that, but I'm not sure how to help. (I already inserted an explanation in a few places, it's not 100% sure this has to be one of them, but it felt relevant).

Comment (ilyagr) [reply docs/git-comparison.md:114]: I'll update the TODO in a moment.

File: docs/glossary.md (hunk 2 above)
Comment (ilyagr) [reply docs/glossary.md:30]: The most important case (which I think has been addressed to a degree recently) is if you have

```
x -> branch_b -> branch_a
 \-> main
```

and you rebase `branch_a` onto `main`, `branch_b` normally does not move.

----

But I think you have a point, I'll reword this a bit to trigger people's sense of "um, actually" a bit less.

File: docs/git-comparison.md (hunk 1 above)
Comment (emilazy) [reply docs/git-comparison.md]: To be clear the outdated part was just that it’s talking about `jj branch` :)

File: docs/glossary.md (hunk 2 above)
Comment (martinvonz) [reply docs/glossary.md]: Good example, but please mention the `--update-refs` flag (which `git rebase` learned about two years ago) so readers don't think we don't know about or that we pretend that it doesn't exist.

File: docs/git-comparison.md (hunk 1 above)
Comment (ilyagr) [reply docs/git-comparison.md]: Ah, great point, thanks!

File: docs/git-comparison.md [hunk 5]
```diff
@@ -243,17 +249,17 @@ parent.
...
       <td><code>jj new main</code></td>
       <td><code>git switch -c topic main</code> or
         <code>git checkout -b topic main</code> (may need to stash or commit
         first)</td>
     </tr>
     <tr>
-      <td>Move bookmark A onto bookmark B</td>
+      <td>Move bookmark/branch A onto bookmark/branch B</td>
       <td><code>jj rebase -b A -d B</code></td>
       <td><code>git rebase B A</code>
```
Comment (ilyagr) [inline docs/git-comparison.md:261]: @martinvonz: Would it be correct to say `git rebase B A --update-refs` here and remove the comment below? Or would you prefer to mention `--update-refs` in the comment? I'd probably put it into a separate commit.

File: docs/glossary.md (hunk 2 above)
Comment (ilyagr) [reply docs/glossary.md]: I just removed "Unlike in Git" from this sentence (and moved it to another sentence).

But see also https://github.com/martinvonz/jj/pull/4654#discussion_r1802344909, which we could also fix.

File: docs/git-comparison.md (hunk 5 above)
Comment (martinvonz) [reply docs/git-comparison.md:261]: No, I don't think that would be correct because it would still not rebase descendants. If you had a linear chain of bookmarks A..C, then `git rebase D C` would rebase A, B, and C. But it would still fail if it was not linear, so I think it makes sense to leave the comment here.

File: docs/glossary.md (hunk 2 above)
Comment (martinvonz) [reply docs/glossary.md]: I think it makes sense to leave it for another commit/PR.

File: docs/git-comparison.md (hunk 5 above)
Comment (ilyagr) [reply docs/git-comparison.md:261]: OK, I'll leave it be then. For some strange reason, I stopped following new Git features a few years ago :)
